app/routes.py

```diff
@@ -1,2 +1,291 @@
-from flask import Blueprint
+from app import db
+from app.models.customer import Customer
+from flask import Blueprint, request, make_response, jsonify
+from datetime import datetime, timedelta
+import requests
+import os
+from app.models.video import Video
+from app.models.rental import Rental
+import flask_migrate 
 
+customer_bp = Blueprint("customer", __name__, url_prefix = "/customers")
+video_bp = Blueprint("video", __name__, url_prefix = "/videos")
+rental_bp = Blueprint("rental", __name__, url_prefix = "/rentals")
+
+# --------------------------CUSTOMER ENDPOINTS--------------------------------
+
+@customer_bp.route("", methods = ["GET"])
+def get_all_customers():
+    customers = Customer.query.all()
+    all_customers = []
+    for customer in customers:
+        all_customers.append(customer.to_json())
+    
+    return jsonify(all_customers), 200
+
+
+@customer_bp.route("/<id>", methods = ["GET"])
+def customer_by_id(id):
+    customer = Customer.query.get(id)
+    if customer == None:
+        return jsonify({"details": "customer not found"}), 404
+
+    else:
+        return jsonify(customer.to_json()), 200
+
+@customer_bp.route("/<id>", methods = ["PUT"])
+def update_customer(id):
+    customer = Customer.query.get(id)
+    request_body = request.get_json()
+    if customer == None:
+        return jsonify({"details": "customer not found"}), 404
```

Comment:
This code is the same as lines 30 & 31 - consider pulling them out into a helper function.  The Flask abort function might be useful in this case, it makes it possible to return a response directly from a helper function.